### PR TITLE
chore: Refactor meta-transaction v2 to handle different meta-transaction types [RFQ-828]

### PR DESCRIPTION
**Pull request by Haozhuo**
# Description

Initially, meta-transaction v2 endpoints are designed to only handle meta-transaction v2 types. However, there is a new use case that needs v2 endpoints to support existing meta-transaction. This PR makes the corresponding changes.

# Testing & Simbot Run

# Checklist

-   [ ] Update 0x API documentation (gitbook) if needed (e.g. public facing API change).
-   [ ] Add tests to cover changes as needed.
-   [ ] All dependent changes (e.g. RFQ server, Gas Price Oracle) have been deployed (if any).

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -245,12 +245,18 @@ export interface MetaTransactionV1QuoteResponse extends BasePriceResponse {
...
  */
 export interface MetaTransactionV2QuoteResponse extends BasePriceResponse {
-    metaTransactionHash: string;
-    // TODO(vic): Update to MetaTransactionV2 when it's ready
-    metaTransaction: ExchangeProxyMetaTransaction;
+    metaTransaction: MetaTransactionField;
     fees?: Fees;
 }
 
+type MetaTransactionField = MetaTransactionV1Field /* add meta-transaction v2 when it's ready */;
```
Comment (Haozhuo) [inline src/types.ts:252]: Need some suggestions on the name of the type @phil-ociraptor @byeongminP

File: src/types.ts [hunk 2]
```diff
@@ -245,12 +245,18 @@ export interface MetaTransactionV1QuoteResponse extends BasePriceResponse {
...
+
+interface MetaTransactionV1Field {
+    kind: 'v1';
+    hash: string;
+    object: MetaTransaction;
```
Comment (Haozhuo) [inline src/types.ts:257]: Same for this field

File: src/types.ts (hunk 1 above)
Comment (byeongminP) [reply src/types.ts:252]: If we were to go for the same nomenclature as `rfq-api`, I'd suggest `MetaTransactionV1Response`

File: src/types.ts [hunk 3]
```diff
@@ -245,12 +245,18 @@ export interface MetaTransactionV1QuoteResponse extends BasePriceResponse {
  * Response type for /meta_transaction/v2/quote endpoint
  */
 export interface MetaTransactionV2QuoteResponse extends BasePriceResponse {
-    metaTransactionHash: string;
-    // TODO(vic): Update to MetaTransactionV2 when it's ready
-    metaTransaction: ExchangeProxyMetaTransaction;
+    metaTransaction: MetaTransactionField;
```
Comment (phil-ociraptor) [inline src/types.ts:248]: can we call this
```suggestion
    trade: TradeContext;
```

**Review by phil-ociraptor — APPROVED** [review]

**Pull request merged**